aioshelly/rpc_device/wsrpc.py

```diff
@@ -323,41 +327,53 @@ def handle_frame(self, frame: dict[str, Any]) -> None:
     async def _rx_msgs(self) -> None:
         assert self._client
 
-        while not self._client.closed:
-            try:
-                msg = await self._client.receive()
-                self._last_time = time.time()
-                if msg.type == WSMsgType.PONG:
-                    self._schedule_heartbeat()
-                    continue
-                if msg.type == WSMsgType.PING:
-                    await self._client.pong(msg.data)
-                    continue
-                frame = _receive_json_or_raise(msg)
-                _LOGGER.debug("recv(%s): %s", self._ip_address, frame)
-            except InvalidMessage as err:
-                _LOGGER.error("Invalid Message from host %s: %s", self._ip_address, err)
-            except ConnectionClosed:
-                break
-            except Exception:  # pylint: disable=broad-except
-                _LOGGER.exception("Unexpected error while receiving message")
+        try:
+            while not self._client.closed:
+                try:
+                    msg = await self._client.receive()
+                    self._last_time = time.time()
+                    if msg.type == WSMsgType.PONG:
+                        self._schedule_heartbeat()
+                        continue
+                    if msg.type == WSMsgType.PING:
+                        await self._client.pong(msg.data)
+                        continue
+                    frame = _receive_json_or_raise(msg)
+                    _LOGGER.debug("recv(%s): %s", self._ip_address, frame)
+                except InvalidMessage as err:
+                    _LOGGER.error(
+                        "Invalid Message from host %s: %s", self._ip_address, err
+                    )
+                except ConnectionClosed:
+                    break
+                except Exception:  # pylint: disable=broad-except
+                    _LOGGER.exception("Unexpected error while receiving message")
+                    raise
```

Comment:
And finally if we did get a RuntimeError on concurrent receive it would infinite loop because it swallowed the exception and would happen again next loop